docs/_docs/04_reference/config-blocks-and-attributes.md

```diff
@@ -1214,6 +1215,64 @@ export TERRAGRUNT_FEATURE=run_hook=true,string_flag=dev
 terragrunt apply
 ```
 
+### exclude
+
+The `exclude` block in Terragrunt provides advanced configuration options to dynamically determine when and how specific
+units in the Terragrunt dependency graph are excluded. This feature allows for fine-grained control over which actions
+are executed and can conditionally exclude dependencies.
+
+Syntax:
+
+```hcl
+exclude {
+    if = <boolean expression>           # Boolean expression to determine exclusion.
+    actions = ["<action>", ...]         # List of actions to exclude (e.g., "plan", "apply", "all", "all_except_output").
+    exclude_dependencies = <boolean>    # Boolean to determine if dependencies should also be excluded.
+}
+```
+
+Attributes:
+
+| Attribute              | Type               | Description                                                                                                             |
+|------------------------|--------------------|-------------------------------------------------------------------------------------------------------------------------|
+| `if`                   | Boolean Expression | A condition to dynamically determine whether the unit should be excluded.                                               |
+| `actions`              | List of Strings    | Specifies which actions to exclude when the condition is met. Options: `plan`, `apply`, `all`, `all_except_output` etc. |
+| `exclude_dependencies` | Boolean            | Indicates whether the dependencies of the excluded unit should also be excluded.                                        |
```

Comment:
This doesn't mention the default value. I'm assuming it's `false`, as that's the bottom value in Golang?

Reply:
Let's explicitly indicate what's required and what isn't.

Reply:
Also, we should document what happens when `exclude_dependencies` is true, but `if` resolves to `false`.